### ISSUE TITLE
Add support for multiple shared memory OSRM datasets

### ISSUE THREAD

**Issue by KieCap** [VROOM-Project/vroom#181]
Since OSRM 5.17, multiple datasets can be loaded into memory at the same time, using:

```
$ ./osrm-datastore --dataset-name=mydata /path/to/data.osrm
$ ./osrm-routed --dataset-name=mydata --shared-memory
```

It'd be nice if VROOM had support for this, as extracting the entire planet for routing is a painstaking task!

It could also probably be used to load multiple profiles in memory and decide on the profile at the VROOM query level.

**Comment by jcoupey** [comment]
It would be good to have this, first to be on par with what options OSRM has to offer. Then it would open up to more flexibility in the integration.

When using `osrm-routed` with several datasets - with or without shared memory - we already have a way to select what instance is triggered by either using different ports along with `-p`, or using the `-m` option for profile (requires to proxy the OSRM requests based on the infamous unused `{profile}` parameter in the OSRM API).

But none of this is doable right now when using `libosrm`. We could probably just reuse the `-m` value (currently unused with `libosrm`) so that it describes the dataset name on a per request basis.

**Comment by jcoupey** [comment]
This landed with #196 and is related to #187, so using different datasets with `libosrm` is just like using different profiles.

Load the data using custom names:
```
./osrm-datastore --dataset-name=car /path/to/car_data.osrm
./osrm-datastore --dataset-name=foot /path/to/foot_data.osrm
```

Run VROOM with `libosrm`:
```
vroom -r libosrm [...]
```

Default behaviour with no change to the input file amounts to firing OSRM requests against the `car` profile/dataset. Hitting the `foot` profile requires to add a `profile: "foot"` key for **all** vehicles in input. Valid `profile` keys are determined by the dataset names loaded in memory with `osrm-datastore`.

@KieCap you can use this approach with any combination of profile/region that result in different datasets, so hopefully this addresses your initial concern of splitting the entire planet extraction.
